packages/@ama-mfe/ng-utils/src/navigation/navigation.producer.service.ts

```diff
@@ -0,0 +1,127 @@
+import type {
+  Navigation,
+  NavigationVersions,
+} from '@ama-mfe/messages';
+import {
+  MessagePeerService,
+} from '@amadeus-it-group/microfrontends-angular';
+import {
+  DestroyRef,
+  inject,
+  Injectable,
+} from '@angular/core';
+import {
+  takeUntilDestroyed,
+} from '@angular/core/rxjs-interop';
+import {
+  ActivatedRoute,
+  NavigationEnd,
+  NavigationStart,
+  Router,
+} from '@angular/router';
+import {
+  filter,
+  map,
+  switchMap,
+} from 'rxjs';
+import {
+  type MessageProducer,
+  ProducerManagerService,
+} from '../managers/index';
+import {
+  type ErrorContent,
+} from '../messages/error';
+
+export interface RoutingServiceOptions {
+  /**
+   * Whether to handle only sub-routes.
+   * If true, the routing service will handle only sub-routes.
+   * Default is false.
+   */
+  subRouteOnly?: boolean;
+}
+
+/**
+ * A service that handles routing and message production for navigation events.
+ *
+ * This service listens to Angular router events and sends navigation messages
+ * to a message peer service. It also handles errors related to navigation messages.
+ */
+@Injectable({
+  providedIn: 'root'
+})
+export class RoutingService implements MessageProducer<NavigationVersions> {
+  private readonly router = inject(Router);
+  private readonly activatedRoute = inject(ActivatedRoute);
+  private readonly messageService = inject(MessagePeerService<NavigationVersions>);
+
+  public readonly types = 'navigation';
+
+  private readonly takeUntilDestroyedOperator = takeUntilDestroyed();
+
+  constructor() {
+    const producerManagerService = inject(ProducerManagerService);
+    producerManagerService.register(this);
+
+    inject(DestroyRef).onDestroy(() => {
+      producerManagerService.unregister(this);
+    });
+  }
+
+  /**
+   * Handles errors related to navigation messages.
+   * @param message - The error message content.
+   */
```

Comment:
```suggestion
  /** @inheritDoc */
```

Would not do the job?